Add files via upload

arjoon-r opened the pull request:
Derive_param_maxsev.R

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue so that it closes after successful merging.
- [ ] Address all merge conflicts and resolve appropriately.
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

bms63 commented on R/derive_param_maxsev_draft.R:
```diff
@@ -0,0 +1,124 @@
+#' Title: DERIVE_PARAM_MAXSEV 
```
Lets remove the title from here

bms63 replied:
Hey Arjun,  Please look over the template for making proper roxygen documentation in the top of the file - https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#function-header-documentation

bms63 commented on R/derive_param_maxsev_draft.R:
```diff
@@ -0,0 +1,124 @@
+#' Title: DERIVE_PARAM_MAXSEV 
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
+#' @description                                                                 +
+#' This derive_param_maxsev function is used to derive the maximum severity for +
+#'  each administrative and Systemic events.                                    + 
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```
```suggestion
#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
#' @description                                                                 +
#' This `derive_param_maxsev` function is used to derive the maximum severity for +
#'  each administrative and Systemic events.                                    + 
#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

```

arjoon-r replied:
Hi Ben, I have updated the code. Please check the latest version.

bms63 commented on R/derive_param_maxsev_draft.R:
```diff
@@ -0,0 +1,124 @@
+  by_var = NULL,
+  period_var = NULL
+){
+  #vars check
```
```suggestion
  # vars check

```

bms63 reviewed (changes requested):
Please see requested changes

bms63 reviewed (changes requested):
Hey Arjun.  Lets just do one function per Pull Reqest.  Also, the Lock File should not change.  I see you updated Description and Readme, but we should realyl do those in separate PRs just to keep it simple

bms63 commented:
Also we need unit tests

arjoon-r commented:
> Hey Arjun. Lets just do one function per Pull Reqest. Also, the Lock File should not change. I see you updated Description and Readme, but we should realyl do those in separate PRs just to keep it simple

Hi Ben, I have removed that new code (derive_var_sev.R). I have updated the derive_param_maxsev.R code as per your comments. Please check the latest version.

bms63 commented:
@pharmaverse/admiralvaccines Hi Team - I fixed a merge conflict that was holding up the CI checks in @arjoon-r 's code.  They checks are now firing off.  Click on details and see what is being flagged as breaking the CI checks.  These are very easy to fix!  

@arjoon-r can you run either `devtools::document()` or `roxygen2::roxygenize('.', roclets = c('rd', 'collate', 'namespace'))` to update the documentation for your function.  This will create an additional `.Rd` file for your function.

yangc71 commented on R/derive_param_maxsev_draft.R:
```diff
@@ -0,0 +1,107 @@
+#'
+#' @author ARJUN R
+#'
+#' @examples
```
Please make sure the examples are fully self-contained as indicated in the [programming strategy](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html): "A fully self-contained example of how to use the function. Self-contained means that, if this code is executed in a new R session, it will run without errors. That means any packages need to be loaded with [library()](https://rdrr.io/r/base/library.html) and any datasets needed either to be created directly inside the example code or loaded using [data()](https://rdrr.io/r/utils/data.html). If a dataset is created in the example, it should be done so using the function [tribble()](https://tibble.tidyverse.org/reference/tribble.html) (specify [library(tibble)](https://tibble.tidyverse.org/) before calling this function). If other functions are called in the example, please specify [library(pkg_name)](https://rdrr.io/r/base/library.html) then refer to the respective function fun() as opposed to the preferred pkg_name::fun() notation as specified in [Unit Test Guidance](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#set-up-the-test-script). Make sure to align columns as this ensures quick code readability."

yangc71 reviewed (commented):
As a general comment for creating pull requests, please  link the relevant issue to the pull reques, and create the pull request with format "Close #issue_number: short description of pull request". Please reference the "Pull Request" section of the [admiraldev git usage guide](https://pharmaverse.github.io/admiraldev/main/articles/git_usage.html#pull-request) for more information.
